avocado/core/runner.py

```diff
@@ -530,16 +530,16 @@ def run_suite(self, test_suite, variants, timeout=0, replay_map=None):
                         summary.add('INTERRUPTED')
                         if 'methodName' in test_parameters:
                             del test_parameters['methodName']
-                        test_factory = (test.TimeOutSkipTest, test_parameters)
+                        test_factory = (test.TimeOutSkipTest, test_parameters,
+                                        None)
                         break_loop = not self.run_test(test_factory, queue,
                                                        summary)
                         if break_loop:
                             break
                     else:
-                        if (replay_map is not None and
-                                replay_map[index] is not None):
-                            test_parameters["methodName"] = "test"
-                            test_factory = (replay_map[index], test_parameters)
+                        if (replay_map is not None and index < len(replay_map)):
+                            test_factory = (test_factory[0], test_factory[1],
+                                            replay_map[index])
```

Comment:
As I said, I don't fancy this third element in the `test_factory`. But I see what you're  trying here. I'd keep all this logic in the runner and keep the `test_factory` as is.